nomos-da/network/core/src/protocols/replication/behaviour.rs

```diff
@@ -213,16 +221,22 @@ pub struct ReplicationBehaviour<Membership> {
     outbound_streams: HashMap<PeerId, WriteHalfState>,
     /// Holds tasks for writing messages to the streams' write halves
     outbound_tasks: FuturesUnordered<OutboundTask>,
-    /// Seen messages cache holds a record of seen messages
-    seen_message_cache: IndexSet<(Vec<u8>, SubnetworkId)>,
+    /// The cache for seen messages, evicts entries in two ways:
+    /// - oldest entries when full (Least-Recently-Used),
+    /// - after a certain TTL has passed since the entry was last accessed,
+    ///   regardless of cache utilization.
+    /// The cache allocates memory once up front and any expired and unused
+    /// entries will eventually be overwritten and erasing them via
+    /// [`TimeSizedCache::flush`] does not save space.
+    seen_message_cache: TimedSizedCache<(Vec<u8>, SubnetworkId), ()>,
```

Comment:
What is `Vec<u8>` in this context? the bytes of the message before deserialization?
Wouldn't it be more clear to use the `blob_id` in the same fashion as used in storage?